aea/test_tools/test_cases.py

```diff
@@ -358,27 +381,108 @@ def _read_error(pid: subprocess.Popen):
         """
         Start a tty reading thread.
 
         :param process: target process passed to a thread args.
 
         :return: None.
         """
-        self.start_thread(target=self._read_tty, process=process)
+        cls.start_thread(target=cls._read_tty, process=process)
 
-    def start_error_read_thread(self, process: subprocess.Popen) -> None:
+    @classmethod
+    def start_error_read_thread(cls, process: subprocess.Popen) -> None:
         """
         Start an error reading thread.
 
         :param process: target process passed to a thread args.
 
         :return: None.
         """
-        self.start_thread(target=self._read_error, process=process)
+        cls.start_thread(target=cls._read_error, process=process)
+
+    @classmethod
+    def _get_cwd(cls) -> str:
+        """Get the current working directory."""
+        return str(cls.t / cls.current_agent_context)
+
+    @classmethod
+    def send_envelope_to_agent(cls, envelope: Envelope, agent: str):
+        """Send an envelope to an agent, using the stub connection."""
+        write_envelope_to_file(envelope, str(cls.t / agent / DEFAULT_INPUT_FILE_NAME))
+
+    @classmethod
+    def read_envelope_from_agent(cls, agent: str) -> Envelope:
+        """Read an envelope from an agent, using the stub connection."""
+        return read_envelope_from_file(str(cls.t / agent / DEFAULT_OUTPUT_FILE_NAME))
 
-    def add_scripts_folder(self):
-        scripts_src = os.path.join(self.cwd, "scripts")
-        scripts_dst = os.path.join(self.t, "scripts")
-        shutil.copytree(scripts_src, scripts_dst)
+    @classmethod
+    def setup_class(cls):
+        """Set up the test class."""
+        cls.runner = CliRunner()
+        cls.old_cwd = Path(os.getcwd())
+        cls.subprocesses = []
+        cls.threads = []
+
+        cls.t = Path(tempfile.mkdtemp())
+        cls.change_directory(cls.t)
+
+        # random.choices(string.ascii_lowercase, 5)
+        cls.registry_tmp_dir = cls.t / "packages"
+        package_registry_src = cls.old_cwd / cls.packages_dir_path
+        shutil.copytree(str(package_registry_src), str(cls.registry_tmp_dir))
+
+        cls.initialize_aea(cls.author)
+
+    @classmethod
+    def teardown_class(cls):
+        """Teardown the test."""
+        cls._terminate_subprocesses()
+        cls._join_threads()
+        cls.unset_agent_context()
+        cls.change_directory(cls.old_cwd)
+        try:
+            shutil.rmtree(cls.t)
+        except (OSError, IOError):
+            pass
+
+
+class UseOef:
+    @pytest.fixture(autouse=True)
+    def _start_oef_node(self, network_node):
+        """Start an oef node."""
+
+
+class AEATestCaseSingle(BaseAEATestCase):
+    """Test case for a single AEA project."""
+
+    agent_name: str
+
+    @classmethod
+    def setup_class(cls):
+        """Set up the test class."""
+        BaseAEATestCase.setup_class()
+        cls.agent_name = "agent-" + "".join(random.choices(string.ascii_lowercase, k=5))
```

Comment:
We need to make sure in the case of testing against a project (see quickstart) that this agent name is taken from `aea-config.yaml`